<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Descripción-del-proyecto" data-toc-modified-id="Descripción-del-proyecto-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Descripción del proyecto</a></span><ul class="toc-item"><li><span><a href="#Descripción-de-los-datos" data-toc-modified-id="Descripción-de-los-datos-1.1"><span class="toc-item-num">1.1&nbsp;&nbsp;</span>Descripción de los datos</a></span></li></ul></li><li><span><a href="#Exploración-de-datos" data-toc-modified-id="Exploración-de-datos-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Exploración de datos</a></span><ul class="toc-item"><li><span><a href="#Importación-de-librerías-y-descarga-de-base-de-datos" data-toc-modified-id="Importación-de-librerías-y-descarga-de-base-de-datos-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>Importación de librerías y descarga de base de datos</a></span></li><li><span><a href="#Exploración-y-tratamiento-de-datos" data-toc-modified-id="Exploración-y-tratamiento-de-datos-2.2"><span class="toc-item-num">2.2&nbsp;&nbsp;</span>Exploración y tratamiento de datos</a></span></li></ul></li><li><span><a href="#Análisis-de-datos" data-toc-modified-id="Análisis-de-datos-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Análisis de datos</a></span><ul class="toc-item"><li><span><a href="#Análisis-de-establecimientos-pertenecientes-a-una-cadena" data-toc-modified-id="Análisis-de-establecimientos-pertenecientes-a-una-cadena-3.1"><span class="toc-item-num">3.1&nbsp;&nbsp;</span>Análisis de establecimientos pertenecientes a una cadena</a></span></li><li><span><a href="#Análisis-de-establecimientos-no-pertenecientes-a-una-cadena." data-toc-modified-id="Análisis-de-establecimientos-no-pertenecientes-a-una-cadena.-3.2"><span class="toc-item-num">3.2&nbsp;&nbsp;</span>Análisis de establecimientos no pertenecientes a una cadena.</a></span></li><li><span><a href="#Análisis-comparativo-de-la-cantidad-de-asientos." data-toc-modified-id="Análisis-comparativo-de-la-cantidad-de-asientos.-3.3"><span class="toc-item-num">3.3&nbsp;&nbsp;</span>Análisis comparativo de la cantidad de asientos.</a></span></li><li><span><a href="#Calles-más-populares" data-toc-modified-id="Calles-más-populares-3.4"><span class="toc-item-num">3.4&nbsp;&nbsp;</span>Calles más populares</a></span></li></ul></li><li><span><a href="#Conclusiones" data-toc-modified-id="Conclusiones-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Conclusiones</a></span></li><li><span><a href="#Presentación" data-toc-modified-id="Presentación-5"><span class="toc-item-num">5&nbsp;&nbsp;</span>Presentación</a></span></li></ul></div>

# Proyecto estudio de mercado para posible apertura de un café.

## Descripción del proyecto

Con la idea de abrir un pequeño café regentado por robots en Los Ángeles, voy a realizar un estudio de mercado en la zona para ver su viabilidad.

Trabajaré con una base de datos proveniente de fuentes abiertas sobre restaurantes en LA.

### Descripción de los datos

El nombre de la tabla es rest_data:

- object_name — nombre del establecimiento
- chain — establecimiento que pertenece a una cadena (TRUE/FALSE)
- object_type — tipo de establecimiento
- address — dirección
- number — número de asientos

A continuación dare comienzo a la exploración del dataset.


## Exploración de datos

### Importación de librerías y descarga de base de datos

In [1]:
# Importación de librerías

import pandas as pd
import plotly.express as px

#Cargando drive, ya que el proyecto fue realizado en Google Colab y el archivo se encuentra en Google Drive
from google.colab import drive
drive.mount('/content/gdrive')

Mounted at /content/gdrive


In [16]:
# Cargar los archivos
try:
  rest_data= pd.read_csv('gdrive/My Drive/Education/Practicum/Sprint 9/Project 9/rest_data_us.csv')

except:
  #leyendo el archivo para el revisor de código de Practicum
  rest_data= pd.read_csv('/datasets/rest_data_us.csv')

In [17]:
rest_data.info()
rest_data.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9651 entries, 0 to 9650
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   id           9651 non-null   int64 
 1   object_name  9651 non-null   object
 2   address      9651 non-null   object
 3   chain        9648 non-null   object
 4   object_type  9651 non-null   object
 5   number       9651 non-null   int64 
dtypes: int64(2), object(4)
memory usage: 452.5+ KB


id          object_name                    address  chain object_type  \
0  11786  HABITAT COFFEE SHOP     3708 N EAGLE ROCK BLVD  False        Cafe   
1  11787             REILLY'S        100 WORLD WAY # 120  False  Restaurant   
2  11788       STREET CHURROS  6801 HOLLYWOOD BLVD # 253  False   Fast Food   
3  11789    TRINITI ECHO PARK         1814 W SUNSET BLVD  False  Restaurant   
4  11790               POLLEN         2100 ECHO PARK AVE  False  Restaurant   

   number  
0      26  
1       9  
2      20  
3      22  
4      20

### Exploración y tratamiento de datos

In [18]:
# Busco valores ausentes en el dataset
rest_data.isna().sum()

id             0
object_name    0
address        0
chain          3
object_type    0
number         0
dtype: int64

In [19]:
# Eliminación de datos ausentes y chequeo de información.

rest_data= rest_data.dropna()
rest_data.info()
rest_data.isna().sum()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9648 entries, 0 to 9650
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   id           9648 non-null   int64 
 1   object_name  9648 non-null   object
 2   address      9648 non-null   object
 3   chain        9648 non-null   object
 4   object_type  9648 non-null   object
 5   number       9648 non-null   int64 
dtypes: int64(2), object(4)
memory usage: 527.6+ KB


id             0
object_name    0
address        0
chain          0
object_type    0
number         0
dtype: int64

In [20]:
# Busco valores duplicados

rest_data.duplicated().sum()

0

In [21]:
# Cambio tipo de dato columna Chain

rest_data['chain']= rest_data['chain'].astype('bool')

# Cambio tipo de dato columna object_type

rest_data['object_type'] = rest_data['object_type'].astype('category')

# Chequeo dataset

rest_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9648 entries, 0 to 9650
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype   
---  ------       --------------  -----   
 0   id           9648 non-null   int64   
 1   object_name  9648 non-null   object  
 2   address      9648 non-null   object  
 3   chain        9648 non-null   bool    
 4   object_type  9648 non-null   category
 5   number       9648 non-null   int64   
dtypes: bool(1), category(1), int64(2), object(2)
memory usage: 395.9+ KB


Una vez terminada la exploración y tratamiento de los datos, comenzaré con el análisis.

## Análisis de datos

In [22]:
# Investigo las proporciones de los distintos tipos de establecimientos en el mercado.

prop_object_type= rest_data.groupby('object_type').agg({'id':'count'}).sort_values(by='id', ascending= False).reset_index()
prop_object_type.columns=['object_type','amount']
prop_object_type



object_type  amount
0  Restaurant    7253
1   Fast Food    1066
2        Cafe     435
3       Pizza     319
4         Bar     292
5      Bakery     283

In [23]:
# Gráfico mostrando la proporción del mercado por tipo de establecimiento.

fig = px.pie(prop_object_type, values="amount", names="object_type", title='Proporción del mercado por tipo de establecimiento')
fig.show()

Podemos observar que la mayor parte del mercado esta ocupado por los restaurantes seguido por los fast food aunque en menores proporciones. Los cafes estan en tercer lugar con tan solo el 4.51% del mercado y rondando el 3% cada uno, se encuentran la pizza, bar y bakery.

In [25]:
# Investigo las proporciones de los establecimientos que pertenecen a una cadena y de los que no.

prop_chain= rest_data.groupby('chain').agg({'id':'count'}).sort_values(by='id', ascending= False).reset_index()
prop_chain.columns= ['chain','amount']
prop_chain

chain  amount
0  False    5972
1   True    3676

In [26]:
# Gráfico mostrando proporciones de los establecimientos que pertenecen a una cadena y los que no.

fig1 = px.pie(prop_chain, values="amount", names="chain", title='Proporción por tipo de establecimiento: ¿Pertenece a una cadena?')
fig1.show()

Podemos observar que el 61.9% de los establecimientos no pertenece a una cadena.

### Análisis de establecimientos pertenecientes a una cadena

In [27]:
# Busco los establecimientos que pertenecen a una cadena.

chain= rest_data.query('chain == [True]')
chain.head()

id               object_name                          address  chain  \
8   11794                ABC DONUTS  3027 N SAN FERNANDO RD UNIT 103   True   
10  11796             EL POLLO LOCO               5319 W SUNSET BLVD   True   
11  11797         POONG NYUN BAKERY        928 S WESTERN AVE STE 109   True   
12  11798  EMC RESTAURANT GROUP LLC            3500 W 6TH ST STE 101   True   
14  11800         CUSCATLECA BAKERY            2501 W SUNSET BLVD #A   True   

   object_type  number  
8    Fast Food       1  
10  Restaurant      38  
11      Bakery      17  
12  Restaurant     133  
14      Bakery      26

In [28]:
# Busco proporciones de los establecimientos que pertenecen a una cadena.

chain_amount= chain.groupby('object_type').agg({'id':'count'}).sort_values(by='id', ascending= False).reset_index()
chain_amount.columns= ['name','amount']
chain_amount

name  amount
0  Restaurant    2292
1   Fast Food     605
2      Bakery     283
3        Cafe     266
4       Pizza     153
5         Bar      77

In [29]:
# Gráfico mostrando las proporciones de los establecimientos que pertenecen a una cadena.

fig2 = px.pie(chain_amount, values="amount", names="name", title='Proporción por tipo de establecimiento perteneciente a una cadena.')
fig2.show()

Como podemos observar, de los establecimientos que pertenecen a una cadena, los restaurantes son los más populares ocupando el 62% del mercado de las cadenas seguido, aunque en una menor proporción, por las fast food con el 16.5%. El 7.24% de las cadenas son cafés.

In [30]:
# Cadenas: muchos establecimientos con un pequeño número de asientos o unos pocos establecimientos con un muchos asientos

chain_model= chain.groupby('object_type').agg({'id':'count','number':'sum'}).sort_values(by='id', ascending= False).reset_index()
chain_model.columns= ['name','amount','seats']
chain_model

name  amount   seats
0  Restaurant    2292  102810
1   Fast Food     605   23044
2      Bakery     283    6162
3        Cafe     266    6894
4       Pizza     153    3906
5         Bar      77    3100

In [31]:
# Busco cantidad de asientos promedio por establecimiento

chain_model['seats per business']= chain_model['seats']/chain_model['amount']
chain_model

name  amount   seats  seats per business
0  Restaurant    2292  102810           44.856021
1   Fast Food     605   23044           38.089256
2      Bakery     283    6162           21.773852
3        Cafe     266    6894           25.917293
4       Pizza     153    3906           25.529412
5         Bar      77    3100           40.259740

A primera vista, se puede percibir un patrón en el cual el incremento en el número de negocios se correlaciona con un aumento en la cantidad de asientos disponibles. Sin embargo, un análisis más profundo revela que las cadenas de establecimientos carecen de una característica uniforme en relación con la relación entre la cantidad de asientos y la cantidad de establecimientos. Aunque existe una leve tendencia a que haya pocos establecimientos con una mayor cantidad de asientos.

Cuando examinamos el segmento de restaurantes, el cual ocupa una porción significativa del mercado de cadenas, es notable que la cantidad de asientos es prácticamente equivalente a la de los bares, que son los establecimientos con menor representación en dicho segmento.

Si nos enfocamos en los bares en comparación con otros tipos de establecimientos (como comida rápida, panaderías, cafeterías y pizzerías), se observa que los bares cuentan con relativamente pocos establecimientos que ofrecen una mayor cantidad de asientos en proporción.

En el caso de las cafeterías, a pesar de tener una mayor presencia en el mercado en comparación con los bares, se destaca que su cantidad de asientos es la mitad de la de estos últimos.

In [32]:
# Representación gráfica de la cantidad de asientos.

fig3= px.bar(chain_model,x='name', y='seats per business', title='Cantidad de asientos promedio por establecimiento pertenecientes a una cadena')
fig3.show()

### Análisis de establecimientos no pertenecientes a una cadena.


In [33]:
# Busco los establecimientos que no pertenecen a una cadena

autonomo= rest_data.query('chain == [False]')
autonomo.head()

id          object_name                    address  chain object_type  \
0  11786  HABITAT COFFEE SHOP     3708 N EAGLE ROCK BLVD  False        Cafe   
1  11787             REILLY'S        100 WORLD WAY # 120  False  Restaurant   
2  11788       STREET CHURROS  6801 HOLLYWOOD BLVD # 253  False   Fast Food   
3  11789    TRINITI ECHO PARK         1814 W SUNSET BLVD  False  Restaurant   
4  11790               POLLEN         2100 ECHO PARK AVE  False  Restaurant   

   number  
0      26  
1       9  
2      20  
3      22  
4      20

In [34]:
# Busco proporciones de los establecimientos que no pertenecen a una cadena.

autonomo_cant= autonomo.groupby('object_type').agg({'id':'count'}).sort_values(by='id', ascending= False).reset_index()
autonomo_cant.columns= ['name','amount']
autonomo_cant

name  amount
0  Restaurant    4961
1   Fast Food     461
2         Bar     215
3        Cafe     169
4       Pizza     166
5      Bakery       0

In [35]:
# Gráfico mostrando las proporciones de los establecimientos que no pertenecen a una cadena.

fig4 = px.pie(autonomo_cant, values="amount", names="name", title='Proporción por tipo de establecimiento que no pertenece a una cadena.')
fig4.show()

La observación revela que, dentro del conjunto de establecimientos que operan de manera independiente sin formar parte de cadenas, los restaurantes ostentan la mayor cuota de popularidad, alcanzando un porcentaje del 83.1%. Les siguen en la jerarquía los establecimientos de comida rápida, con un 7.72%, mientras que en proporciones prácticamente equivalentes, alrededor del 3%, se encuentran los bares, las cafeterías y las pizzerías.

Cabe señalar que, en el contexto de este análisis, no se registra ninguna presencia de establecimientos de tipo panadería.

In [36]:
# Calculo número de asientos por establecimiento

autonomo_model= autonomo.groupby('object_type').agg({'id':'count','number':'sum'}).sort_values(by='id', ascending= False).reset_index()
autonomo_model.columns= ['name','amount','seats']
autonomo_model

name  amount   seats
0  Restaurant    4961  245688
1   Fast Food     461   10895
2         Bar     215    9972
3        Cafe     169    3981
4       Pizza     166    5200
5      Bakery       0       0

In [37]:
# Busco cantidad de asientos promedio por negocio

autonomo_model['seats per business']= autonomo_model['seats']/autonomo_model['amount']
autonomo_model

name  amount   seats  seats per business
0  Restaurant    4961  245688           49.523886
1   Fast Food     461   10895           23.633406
2         Bar     215    9972           46.381395
3        Cafe     169    3981           23.556213
4       Pizza     166    5200           31.325301
5      Bakery       0       0                 NaN

Al observar la tabla, el restaurante es el establecimiento con más establecimientos y más asientos con un promedio de 49. Le sigue los bares. A pesar de tener muchos menos establecimientos tienen un promedio de 46 asientos por lugar. A pesar de no tener tanta dferencia en cantidad de negocios entre los cafes y los bares, los cafés tienen la mitad de asientos que los bares.

En proporción los establecimientos tienen menos lugares pero más asientos.

In [38]:
# Representación gráfica de la cantidad de asientos.

fig5= px.bar(autonomo_model,x='name', y='seats per business', title='Cantidad de asientos promedio por establecimiento que no pertenece a una cadena')
fig5.show()



### Análisis comparativo de la cantidad de asientos.

A contnuación voy a mostrar la cantidad de asientos por establecimiento tanto para aquellos pertenecientes a una cadena como a aquellos que trabajan de manera independiente.

In [39]:
# Uno las tablas de los establecimientos pertenecientes a una cadena y los independientes para comparar cantidad de asientos.

union= pd.merge(chain_model,autonomo_model, on='name')
union.columns=['name','amount_chain','seats_chain','seats_per_business_chain', 'amount_autonomo','seats_autonomo','seats_per_business_independent']
union

name  amount_chain  seats_chain  seats_per_business_chain  \
0  Restaurant          2292       102810                 44.856021   
1   Fast Food           605        23044                 38.089256   
2      Bakery           283         6162                 21.773852   
3        Cafe           266         6894                 25.917293   
4       Pizza           153         3906                 25.529412   
5         Bar            77         3100                 40.259740   

   amount_autonomo  seats_autonomo  seats_per_business_independent  
0             4961          245688                       49.523886  
1              461           10895                       23.633406  
2                0               0                             NaN  
3              169            3981                       23.556213  
4              166            5200                       31.325301  
5              215            9972                       46.381395

In [40]:
# Grafico cantidad de asientos por establecimieno.

fig6= px.bar(union,x='name', y=['seats_per_business_chain','seats_per_business_independent'], title='Seats per business')
fig6.show()

Al observar el gráfico, notamos que el rubro que tiene mayor cantidad de asientos promedio es el de restaurantes. Seguido por el bar.

Los establecimientos independientes, tienen más asientos en los rubros de restaurant, pizza y bar mientras que los establecimientos que pertenecen a una cadena tienen mayor antidad de asientos en los fast food, cafe.

Con respecto al rubro bakery, podemos observar que los establecimientos que hay, integran alguna cadena. No hay ninguna independiente.

### Calles más populares

A continuación voy a buscar las diez mejores calles por número de restaurantes y las calles que tienen solo un local. Para ello, primero separare el nombre de las calles y las pondre en una nueva columna. Dado que hay varias calles que hacen distinción entre N,S,E,W voy a dejarlo porque a pesar de ser la misma calle puede que de un lado se den más un tipo de establecimientos que otros.


In [41]:
# Coloco los datos de los nombres de las calles de la columna address en una columna separada.

# PASO 1:
# Todas las direcciones tienen un número hacia el extremo izquierdo. Vamos a eliminar ese número
nuevas_direcciones = rest_data['address']
nuevas_direcciones = nuevas_direcciones.str.replace("^[0-9]* ", "",regex=True)


# PASO 2:
# Algunas direcciones tienen números al final. Vamos a quitarlos
nuevas_direcciones = nuevas_direcciones.str.replace(" [0-9]*$", "",regex=True)
nuevas_direcciones = nuevas_direcciones.str.replace("[0-9]*$", "",regex=True)

# PASO 3:
# Nos quedan algunos símbolos de # al final de algunos strings
nuevas_direcciones = nuevas_direcciones.str.replace(" #", "",regex=True)
nuevas_direcciones

# Paso 4:
# Coloco los datos en la columna street
rest_data['street'] = nuevas_direcciones

rest_data


id          object_name                      address  chain  \
0     11786  HABITAT COFFEE SHOP       3708 N EAGLE ROCK BLVD  False   
1     11787             REILLY'S          100 WORLD WAY # 120  False   
2     11788       STREET CHURROS    6801 HOLLYWOOD BLVD # 253  False   
3     11789    TRINITI ECHO PARK           1814 W SUNSET BLVD  False   
4     11790               POLLEN           2100 ECHO PARK AVE  False   
...     ...                  ...                          ...    ...   
9646  21432      HALL OF JUSTICE             217 W TEMPLE AVE  False   
9647  21433          FIN-MELROSE             5750 MELROSE AVE  False   
9648  21434          JUICY WINGZ          6741 HOLLYWOOD BLVD   True   
9649  21435      MEDIDATE COFFEE      548 S SPRING ST STE 100  False   
9650  21436         CAFE SPROUTS  1300 S SAN PEDRO ST STE 111   True   

     object_type  number              street  
0           Cafe      26   N EAGLE ROCK BLVD  
1     Restaurant       9           WORLD WAY  
2      Fast Food      20      HOLLYWOOD BLVD  
3     Restaurant      22       W SUNSET BLVD  
4     Restaurant      20       ECHO PARK AVE  
...          ...     ...                 ...  
9646  Restaurant     122        W TEMPLE AVE  
9647  Restaurant      93         MELROSE AVE  
9648   Fast Food      15      HOLLYWOOD BLVD  
9649        Cafe       6     S SPRING ST STE  
9650  Restaurant      19  S SAN PEDRO ST STE  

[9648 rows x 7 columns]

In [42]:
# Calculo el número de restaurantes por calle.

rest_por_calle= rest_data.groupby('street').agg({'object_type':'count'}).sort_values(by='object_type',ascending=False).reset_index()
rest_por_calle.columns= ['street','amount_of_business']
rest_por_calle

street  amount_of_business
0             W SUNSET BLVD                 305
1               W PICO BLVD                 305
2             WILSHIRE BLVD                 223
3            HOLLYWOOD BLVD                 194
4         SANTA MONICA BLVD                 191
...                     ...                 ...
1813  PASEO RANCHO CASTILLA                   1
1814   PASEO DE LA PLAZA ST                   1
1815      PASADENA AVE STEA                   1
1816       PASADENA AVE STE                   1
1817         hollywood BLVD                   1

[1818 rows x 2 columns]

In [43]:
# Busco el número de calles que tienen solo un local.

calles_poco_populares= rest_por_calle.query('amount_of_business == 1')
calles_poco_populares['amount_of_business'].count()

1193

Podemos observar que hay 1193 calles que tienen solo un local.

In [44]:
# Busco las diez mejores calles por número de establecimientos.

mejores_calles= rest_por_calle.head(10)
mejores_calles

street  amount_of_business
0      W SUNSET BLVD                 305
1        W PICO BLVD                 305
2      WILSHIRE BLVD                 223
3     HOLLYWOOD BLVD                 194
4  SANTA MONICA BLVD                 191
5      S WESTERN AVE                 183
6           W 3RD ST                 173
7      S FIGUEROA ST                 173
8      S VERMONT AVE                 165
9     W OLYMPIC BLVD                 155

In [45]:
# Grafico las diez mejores calles por número de establecimientos.

fig7= px.bar(mejores_calles, x='street', y='amount_of_business', title='Diez mejores calles por número de establecimientos')
fig7.show()

Podemos observar que W SUNSET BLVD y W PICO BLVD tienen la misma cantidad de locales. A partir de WILSHIRE BLVD la cantidad de locales comienza a descender lentamente.

In [46]:
#Busco el número de asientos por calle.

seats_per_street= rest_data.groupby('street').agg({'object_type':'count','number':'sum'}).sort_values(by='object_type',ascending=False).reset_index().head(10)
seats_per_street.columns= ['street','number_of_restaurants','seats']
seats_per_street

street  number_of_restaurants  seats
0      W SUNSET BLVD                    305  15494
1        W PICO BLVD                    305  12342
2      WILSHIRE BLVD                    223  12515
3     HOLLYWOOD BLVD                    194  10808
4  SANTA MONICA BLVD                    191   6840
5      S WESTERN AVE                    183   7745
6           W 3RD ST                    173   7062
7      S FIGUEROA ST                    173   8276
8      S VERMONT AVE                    165   7480
9     W OLYMPIC BLVD                    155   8199

In [47]:
# Grafico la cantidad de asientos en las diez mejores calles.

fig8= px.bar(seats_per_street, x='street', y='seats', title='Distribución de asientos en las diez mejores calles')
fig8.show()

Como se puede inferir, se observa una correspondencia directa entre la densidad de establecimientos y la disponibilidad de asientos en la calle. Sin embargo, en el conjunto de las demás calles, la relación entre la cantidad de asientos y la posición en términos de la cantidad de establecimientos no es uniforme.

Por ejemplo, en la situación específica del Boulevard Wilshire, esta calle ocupa el tercer puesto en términos de la cantidad de establecimientos presentes, pero se posiciona en el segundo lugar en cuanto a la cantidad total de asientos.

De manera análoga, se presenta el caso del Boulevard Santa Monica, que a pesar de figurar en el quinto puesto en la lista de calles según la cantidad de locales, registra la menor cantidad de asientos en comparación con las otras calles analizadas.

Como la idea principal de este análisis es conocer el mercado para abrir un café, buscaré las 10 calles con mayor presencia de cafés.

In [48]:
# Busco las 10 calles con mayor presencia de cafés.

cafe=rest_data.query('object_type=="Cafe"')
cafe_streets= cafe.groupby('street').agg({'object_type':'count','number':'sum'}).sort_values(by='object_type',ascending= False).reset_index().head(10)
cafe_streets.columns= ['street','number_cafe','number_of_seats']
cafe_streets

street  number_cafe  number_of_seats
0      W SUNSET BLVD           21              520
1          WORLD WAY           15              323
2        W PICO BLVD           11              215
3      WILSHIRE BLVD           11              233
4  SANTA MONICA BLVD           10              219
5     HOLLYWOOD BLVD            8              168
6      WESTWOOD BLVD            7              146
7        WORLD WAY T            7              148
8  WILSHIRE BLVD STE            7              223
9        S GRAND AVE            6              107

In [49]:
fig9= px.bar(cafe_streets, x='street', y='number_cafe', title='Calles con mayor presencia de cafés')
fig9.show()

In [50]:
# Grafico la cantidad de asientos en las diez mejores calles.

fig10= px.bar(cafe_streets, x='street', y='number_of_seats', title='Distribución de asientos en las diez mejores calles de café')
fig10.show()

Como es evidente, se puede notar que la calle W SUNSET BLVD sobresale como la vía con la mayor cantidad de establecimientos de cafeterías, al mismo tiempo que ostenta el mayor número de asientos disponibles. A continuación, en términos de esta misma característica, se encuentra la calle WORLD WAY.

En el contexto de WESTWOOD BLVD, WORLD WAY T y WILSHIRE BLVD STE, estas tres calles comparten una cantidad idéntica de establecimientos en forma de locales. No obstante, es importante señalar que WILSHIRE BLVD STE se distingue por contar con una cantidad superior de asientos en comparación con las otras dos calles.

Durante el análisis del mercado, pudimos observar que los bares son una buena opción para tener ya que tiene muy pocos locales en el mercado ( no hay tanta competencia) pero en promedio tienen casi la misma cantidad de asientos que los restaurantes que son los establecimientos que tienen mayor porción del mercado.

A continuación mostrare cuales son las calles con mayor presencia de bares y como se distribuyen los asientos en las  mismas.

In [51]:
# Busco las calles con mayor presencia de bares

bar=rest_data.query('object_type=="Bar"')
bar_streets= bar.groupby('street').agg({'object_type':'count','number':'sum'}).sort_values(by='object_type',ascending= False).reset_index().head(10)
bar_streets.columns= ['street','bar_amount','number_of_seats']
bar_streets

street  bar_amount  number_of_seats
0   W SUNSET BLVD          13              507
1        W 7TH ST          11              659
2     W PICO BLVD          10              548
3  HOLLYWOOD BLVD           9              476
4       WORLD WAY           9              293
5   WILSHIRE BLVD           7              308
6     S FLOWER ST           6              201
7     MELROSE AVE           6              288
8     S SPRING ST           5              235
9     N GRAND AVE           5              289

In [52]:
# Grafico las calles con mayor presencia de bares

fig11= px.bar(bar_streets, x='street', y='bar_amount', title='Calles con mayor presencia de bares')
fig11.show()

In [53]:
# Grafico la cantidad de asientos en las diez mejores calles.

fig12= px.bar(bar_streets, x='street', y='number_of_seats', title='Distribución de asientos en las diez mejores calles de bares')
fig12.show()

La observación permite apreciar que la vía con la mayor capacidad de asientos se posiciona en segundo lugar en términos de la cantidad de bares presentes en ella. En contraste, la calle S FLOWER ST se destaca por poseer la menor cantidad de asientos, aun cuando no figura como la calle con la menor cantidad de establecimientos en su totalidad.

## Conclusiones

En resumen, se evidencia que el mercado en general está compuesto mayormente por establecimientos independientes, los cuales abarcan un 61% de la cuota del mercado. La posibilidad de iniciar una cadena podría presentar ventajas debido a la relativa baja presencia de dichos establecimientos.

En relación a los cafés, su participación en el mercado general es limitada, ocupando un 4%. Sin embargo, se observa una mayor proporción de cafés pertenecientes a cadenas en comparación con los negocios independientes. Dentro de las cadenas, los cafés ocupan un 7%, mientras que en los negocios independientes su representación es de tan solo un 2.83%. Curiosamente, la cantidad promedio de asientos en cafés es casi equivalente en ambos tipos de establecimientos.

En cuanto a los bares, parecen ser una opción atractiva a considerar. Su presencia en el mercado es modesta, lo que sugiere un nicho relativamente poco saturado y con poca competencia. Un dato destacable es que, a pesar de su baja participación, los bares poseen una cantidad reducida de locales, incluso menos que los cafés, pero mantienen una cifra de asientos promedio similar a la de los restaurantes, que son el segmento más presente en el mercado. Esto implica que incluso con un solo negocio de este tipo, se podría llegar a igualar la cantidad de asientos de un restaurante.

En el análisis de las calles más populares, se observa una coincidencia entre la calle con la mayor cantidad de establecimientos (W SUNSET BLVD) y la calle con mayor número de cafés y asientos para estos. Aunque se reconoce la alta afluencia y presencia de cafés en dicha calle, es válido considerar otras calles populares que no cuenten con una presencia tan marcada de cafés y asientos, como podría ser el caso de S GRAND AVE.

La calle W SUNSET BLVD también refleja una notable popularidad en el segmento de bares, aunque no alberga tantos asientos como W 7TH ST, que, a pesar de tener menos locales, ostenta una mayor capacidad de asientos. Además, se aprecia que S FLOWER ST presenta una buena cantidad de bares, aunque la cantidad de asientos disponibles en esta calle es menor en comparación con otras vías.

## Presentación

Presentation: <https://drive.google.com/file/d/1bX6JA3RjseZlywHFKcfbgtvPVr43ZGck/view?usp=sharing>